epoxy-adapter/src/main/java/com/airbnb/epoxy/stickyheader/StickyHeaderLinearLayoutManager.kt

```diff
@@ -43,20 +43,25 @@ class StickyHeaderLinearLayoutManager @JvmOverloads constructor(
 
     override fun onAttachedToWindow(recyclerView: RecyclerView) {
         super.onAttachedToWindow(recyclerView)
-        recyclerView.adapter?.let(this::setAdapter)
+        setAdapter(recyclerView.adapter)
     }
 
     override fun onAdapterChanged(oldAdapter: RecyclerView.Adapter<*>?, newAdapter: RecyclerView.Adapter<*>?) {
         super.onAdapterChanged(oldAdapter, newAdapter)
-        newAdapter?.let(this::setAdapter)
+        setAdapter(newAdapter)
     }
 
     @Suppress("UNCHECKED_CAST")
-    private fun setAdapter(newAdapter: RecyclerView.Adapter<*>) {
-        if (::adapter.isInitialized) adapter.unregisterAdapterDataObserver(headerPositionsObserver)
-        adapter = newAdapter as BaseEpoxyAdapter
-        adapter.registerAdapterDataObserver(headerPositionsObserver)
-        headerPositionsObserver.onChanged()
+    private fun setAdapter(newAdapter: RecyclerView.Adapter<*>?) {
+        adapter?.unregisterAdapterDataObserver(headerPositionsObserver)
+        if (newAdapter is BaseEpoxyAdapter) {
+            adapter = newAdapter
+            adapter?.registerAdapterDataObserver(headerPositionsObserver)
+            headerPositionsObserver.onChanged()
+        } else {
+            adapter = null
+            headerPositions.clear()
```

Comment:
`headerPositionsObserver.onChanged()` doesn't need to be called when clearing the positions?

Reply:
It essentially clears the `headerPositions` and does some extra things which are not necessary when the adapter is null. Hence it's safe to just directly call `headerPositions.clear()`